cmd/pin.go

```diff
@@ -0,0 +1,54 @@
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/marcosnils/bin/pkg/config"
+	"github.com/spf13/cobra"
+)
+
+type pinCmd struct {
+	cmd *cobra.Command
+}
+
+func newPinCmd() *pinCmd {
+	root := &pinCmd{}
+
+	cmd := &cobra.Command{
+		Use:           "pin [<name> | <paths...>]",
+		Short:         "Pins current version of the binaries",
+		SilenceUsage:  true,
+		Args:          cobra.MinimumNArgs(1),
+		SilenceErrors: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg := config.Get()
+
+			binaryList := make(map[string]*config.Binary)
+			for _, b := range cfg.Bins {
+				binaryList[b.RemoteName] = b
```

Comment:
This should use `getBinPath`  similar to update here https://github.com/marcosnils/bin/blob/91ae3e63bc1e2a499a832ba5f4e44d3c071b1345/cmd/update.go?plain=1#L58 so `pin` can be called either by the binary name or the path.